### PR TITLE
[RFC] redirector: shorten known URLs

### PR DESCRIPTION
**Pull request by Tilka**
The design is a bit hacky and I'm a Go noob, suggestions welcome.

### CODE REVIEW

**Comment by MaJoRoesch** [comment]
Dolp.in? As in Dolpin? Why not Dolph.in?

File: redirector/main.go [hunk 1]
```diff
@@ -73,6 +98,11 @@ func Router(w http.ResponseWriter, req *http.Request) {
 		}
 	}
 
+	if req.URL.Path == "/shorten" {
```
Comment (delroth) [inline redirector/main.go]: I would rather this use the HTTP method to differentiate.

File: redirector/main.go [hunk 2]
```diff
@@ -1,48 +1,78 @@
...
-	Redirector{`/i(\d+)/(\d+)/?`, HandleIssueComment},
-	Redirector{`/pr/?(\d+)/?`, MakeStaticRedirector("https://github.com/dolphin-emu/dolphin/pull/")},
-	Redirector{`/pr(/.*)?`, MakeStaticRedirector("https://github.com/dolphin-emu/dolphin/pulls")},
-	Redirector{`/dl(/.*)?`, MakeStaticRedirector("https://dolphin-emu.org/download/")},
-	Redirector{`/gh(/.*)?`, MakeStaticRedirector("https://github.com/dolphin-emu/dolphin")},
-	Redirector{`/git(/.*)?`, MakeStaticRedirector("https://github.com/dolphin-emu/dolphin")},
-	Redirector{`/faq(/.*)?`, MakeStaticRedirector("https://dolphin-emu.org/docs/faq/")},
-	Redirector{`/bbs(/.*)?`, MakeStaticRedirector("https://forums.dolphin-emu.org/")},
+var redirects = []Redirect{
+	Compile(RawRedirect{"/r%s/%s", "https://github.com/dolphin-emu/dolphin/commit/%s#commitcomment-%s", []string{`([0-9a-fA-F]{6,40})`, `(\d+)`}, []string{"$1", "$2"}}),
```
Comment (delroth) [inline redirector/main.go]: Really needs some comments to explain wtf is going on with the %s and the $1 and the regexp parts and ...

**Comment by phire** [comment]
> Dolp.in? As in Dolpin? Why not Dolph.in?

Programmer joke, in regexes, a fullstop matches any letter.

**Comment by delroth** [comment]
Also good luck trying to find a short domain name that matches a dictionary word.

**Comment by delroth** [comment]
As another comment on this: I'd like to keep the possibility to define actual functions for both matching and reverse matching. We'll want something like that when we have some more complex mappings than just regexps (example: some shortening of arbitrary URLs that get an ID associated to them w/ lookup in a DB).

If you can fix that + add some explanation as to how the regexp-based stuff works, I think it will be a really awesome change.

**Comment by Tilka** [comment]
Closing because it's currently too ugly.

**Pull request closed without merging**